iceoryx_posh/include/iceoryx_posh/experimental/popo/sample.hpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_EXPERIMENTAL_POSH_POPO_SAMPLE_HPP
+#define IOX_EXPERIMENTAL_POSH_POPO_SAMPLE_HPP
+
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+#include "iceoryx_utils/cxx/unique_ptr.hpp"
+
+namespace iox
+{
+namespace popo
+{
+
+template <typename T>
+class PublisherInterface;
+
+///
+/// @brief The Sample class is a mutable abstraction over types which are written to loaned shared memory.
+/// These samples are publishable to the iceoryx system.
+///
+template <typename T>
+class Sample
+{
+  public:
+    Sample(cxx::unique_ptr<T>&& samplePtr, PublisherInterface<T>& publisher);
+    /// Creates an empty sample.
+    Sample(std::nullptr_t) noexcept;
+    Sample(const Sample<T>&) = delete;
+    Sample<T>& operator=(const Sample<T>&) = delete;
+    Sample<T>& operator=(Sample<T>&& rhs);
+    Sample(Sample<T>&& rhs);
+    ~Sample();
+
+    ///
+    /// @brief operator =nullptr Clears the sample.
```

Comment:
We should not use this approach to clear a sample. The STL uses `reset` in the unique ptr for that task: https://en.cppreference.com/w/cpp/memory/unique_ptr 
Could you please implement and use `reset` instead of this assignment operator.

Reply:
Reset doesn't make sense here. In the `unique_ptr` it is used to either reset the underlying raw pointer, or as you mentioned, to clear it if no arguments are provided.
Sample instances aren't intended to be resetable (e.g. to a different memory chunk), but they can be "clearable".

I could implement a `clear` method instead, but why do we not want to allow syntax like `sample = nullptr` ? Is it because the sample isn't actually a pointer ?